src/openscm_twolayermodel/base.py

```diff
@@ -0,0 +1,265 @@
+"""
+Module containing the base for model implementations
+"""
+from abc import ABC, abstractmethod
+
+import numpy as np
+import pandas as pd
+import pint
+import pint.errors
+import tqdm.autonotebook as tqdman
+from openscm_units import unit_registry as ur
+from scmdata.run import ScmRun
+from scmdata.timeseries import TimeSeries
+
+from .errors import UnitError
+
+
+class Model(ABC):
+    """
+    Base class for model implementations
+    """
+
+    _save_paras = tuple()  # parameters to save when doing a run
+
+    _name = None  # model name
+
+    @staticmethod
+    def _check_is_pint_quantity(quantity, name, model_units):
```

Comment:
I'd probably call this
`_is_pint_quantity` or `_check_if_pint_quantity`

Reply:
Have updated, resolve if happy